### PR TITLE
Issues/827 front end testing

### PR DESCRIPTION
**Pull request by tstachl**
Loads of stuff in this pull request.

### CODE REVIEW

File: src/commands/scmt/test/run.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+// or any library that is using the messages framework can also be loaded this way.
+const messages = core.Messages.loadMessages('@desklabs/scmt', 'test');
+// const FIELDS = ['AccountRecordsFailed__c', 'AccountRecordsMigrated__c', 'AccountRecordsTotal__c', 'ArticleRecordsFailed__c', 'ArticleRecordsMigrated__c', 'ArticleRecordsTotal__c', 'AttachmentRecordsFailed__c', 'AttachmentRecordsMigrated__c', 'AttachmentRecordsTotal__c', 'CaseRecordsFailed__c', 'CaseRecordsMigrated__c', 'CaseRecordsTotal__c', 'ContactRecordsFailed__c', 'ContactRecordsMigrated__c', 'ContactRecordsTotal__c', 'InteractionRecordsFailed__c', 'InteractionRecordsMigrated__c', 'InteractionRecordsTotal__c', 'NoteRecordsFailed__c', 'NoteRecordsMigrated__c', 'NoteRecordsTotal__c', 'TopicRecordsFailed__c', 'TopicRecordsMigrated__c', 'TopicRecordsTotal__c', 'RecordsFailed__c', 'RecordsMigrated__c', 'RecordsTotal__c', 'Config__c', 'Desk_Config_Id__c', 'Desk_Endpoint__c', 'Error__c', 'JobId__c', 'Log__c', 'Mapping__c', 'Migration_Type__c', 'Object__c', 'StartDate__c', 'Status__c'];
+
+export default class Run extends CommandBase {
```
Comment (kylehaver) [inline src/commands/scmt/test/run.ts:13]: I'm showing a problem in vscode with this line.  

`{
	"resource": "/Users/khaver/code/salesforce/sfdx-plugin-scmt/src/commands/scmt/test/run.ts",
	"owner": "typescript",
	"code": "2417",
	"severity": 8,
	"message": "Class static side 'typeof Run' incorrectly extends base class static side 'typeof CommandBase'.\n  Types of property 'flagsConfig' are incompatible.\n    Type '{ endpoint: IOptionFlag<string>; username: IOptionFlag<string>; password: IOptionFlag<string>; mi...' is not assignable to type 'SfdxFlagsConfig'.\n      Property 'endpoint' is incompatible with index signature.\n        Type 'IOptionFlag<string>' is not assignable to type 'string | boolean | IBooleanFlag<any> | IOptionFlag<any> | JsonMap'.\n          Type 'IOptionFlag<string>' is not assignable to type 'JsonMap'.\n            Index signature is missing in type 'IOptionFlag<string>'.",
	"source": "ts",
	"startLineNumber": 13,
	"startColumn": 22,
	"endLineNumber": 13,
	"endColumn": 25
}`

Comment (tstachl) [reply src/commands/scmt/test/run.ts:13]: @kylehaver does it give a problem compiling or running?

**Comment by kylehaver** [comment]
The Article migration worked perfectly.  However, I ran into an issue when running the case migration in StepThree.  For contacts, the plugin maps the desk field `Desk External Id` to the Salesforce `First Name` field which causes the migration to fail after it is started.  

image

I am testing with the zzz-scktmkb.desk.com site

File: src/commands/scmt/test/run.ts (hunk 1 above)
Comment (Skeyelab) [reply src/commands/scmt/test/run.ts:13]: I have the same error, but it runs.

File: src/commands/scmt/test/run.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  public static description = messages.getMessage('runCommandDescription');
+  public static examples = [
+    `$ sfdx scmt:test:run`,
+    `$ sfdx scmt:test:run -e https://domain.desk.com -n user@domain.com -p password`
```
Comment (Skeyelab) [inline src/commands/scmt/test/run.ts]: I fixed this example.  Other than that, LGTM

**Review by Skeyelab — DISMISSED** [review]
retracted.  I am getting an error on cases

**Comment by Skeyelab** [comment]
I am getting an error on mapping of cases.  

Here is a screen recording: http://cloudapp.skeyelab.com/2o3H2N013o1k

`ERROR running Run:  TypeError: Cannot read property 'y' of null`

**Comment by tstachl** [comment]
@Skeyelab where does the error occur? I can't see it in your recording.

**Comment by tstachl** [comment]
@Skeyelab please review again.

**Review by kylehaver — APPROVED** [review]
The changes since I last reviewed look good.  Tested article and kb migration and output folder was created and sshots taken.

**Pull request merged**